sched/distributor/poolDistributor_test.go

```diff
@@ -0,0 +1,68 @@
+package distributor
+
+import (
+	"github.com/golang/mock/gomock"
+	"github.com/scootdev/scoot/sched"
+	cm "github.com/scootdev/scoot/sched/clustermembership"
+	"testing"
+)
+
+func TestPoolDistributor_NilCluster(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	clusterMock := cm.NewMockCluster(mockCtrl)
+	clusterMock.EXPECT().Members().Return(nil)
+
+	dist := NewPoolDistributor(clusterMock)
+
+	if nil != dist {
+		t.Error("Expected dist to be nil when no members in Cluster")
+	}
+}
+
+func TestPoolDistributor_EmptyCluster(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	nodes := make([]cm.Node, 0)
+
+	clusterMock := cm.NewMockCluster(mockCtrl)
+	clusterMock.EXPECT().Members().Return(nodes)
+
+	dist := NewPoolDistributor(clusterMock)
+
+	if nil != dist {
+		t.Error("Expected dist to be nil when no members in Cluster")
+	}
+}
+
+func TestPoolDistributor_OneNodeCluster(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	nodes := cm.GenerateTestNodes(1)
+	clusterMock := cm.NewMockCluster(mockCtrl)
```

Comment:
AIUI, mocks are used to test interaction with dependencies and stubs/fakes are used when we only care about the result. Here we're testing two things: that Members() is called and that that NewPoolDistributor() succeeds with >1 node in the cluster. If we want to refactor so that NewPoolDistributor() calls a new function AllMembers(), all the tests here have to change. With a fake, we only have to update the MakeFake() implementation, and the tests remain unchanged since they don't care what underlying method actually gets called.

If this isn't a convincing example, I'd like to know why so I can either change my views are explain it differently. Dan, what's the basis of your opinion?

The tests in this PR are simple enough that I don't think the type of test makes a huge difference, but that can change as our code grows.